### DISCUSSION TITLE
AI bot that can play with other players

### DISCUSSION THREAD

**Discussion by 0xlucyy** [aiogram/aiogram#1553] [General]
### aiogram version

3.x

### Problem

I have built a `Telegram Gaming Bot (TGB)` where users can play games against each other. Each game can contain up to 50 players.

Sometimes though, there are slow periods where only 1 user wants to play. 

For example, my TGB receives commands from user_1 to start a game. A game is proposed and joining is opened up to users. A proposed game can only begin when 2 or more users join the game, there is a 15 second timer for others to join the game.

Problem: I started building an `AI bot (AIB)` that can play against users. So, user_1 proposes a game by prompting TGB. Then TBG waits 15 seconds for at least 1 other player.

The problem is that bots cannot see commands/messages from other bots. Therefore, I have no way of getting my AIB to join the proposed game, because joining is done by prompting TGB...

Are there are possible solutions to this?

### Possible solution

I would like some mechanism so that my bots can see each others command, so my AIB can join a game by TGB.

**Comment by Olegt0rr** [comment]
## First solution (cheap)

Imitate your fake-user activities via bot (without real client side), if it fits to your game flow

## Second solution (expensive)

Use libraries `telethon` or `pyrogram` to create actions from real account

**Reply by 0xlucyy** [reply]
Thanks for the reply. How do I `Imitate a fake user with my AI bot`?

**Reply by Olegt0rr** [reply]
I don't know your game flow. Just mock activity of user that not exists

**Reply by 0xlucyy** [reply]
I have an idea, perhaps I can create a mock user that is controlled by the AI bot. Then this mock user can interact with my Telegram Gaming bot..

Do you think that could work??

How do I create a mock user? Do I need another number?

Users just issue commands that the Telegram Gaming Bot recognizes such as /join, /bet, /vote.

**Reply by Olegt0rr** [reply]
Mock - is when your own code imitating real user (first solution)
Another number - is a second solution

**Reply by 0xlucyy** [reply]
Thank you for your input! I will look into both options!

Cheers
